src/routing.js

```diff
@@ -56,24 +56,34 @@ export const parsePath = (pathname, search = '') => {
 
   // App
   const rawParams = search && search.split('?params=')[1]
-  const params = rawParams ? JSON.parse(decodeURIComponent(rawParams)) : null
+  let params = null
+  if (rawParams) {
+    try {
+      params = decodeURIComponent(rawParams)
+    } catch (err) {
+      console.log('The “params” URL parameter is not valid.')
+    }
+  }
+
   const [dao, instanceId, ...appParts] = parts
-  return {
+
+  const completeLocator = {
     ...locator,
     mode: 'app',
     dao,
     instanceId: instanceId || 'home',
     params,
     parts: appParts,
   }
+
+  return completeLocator
 }
 
 // Return a path string for an app instance
 export const getAppPath = ({ dao, instanceId = 'home', params } = {}) => {
+  const paramsPart = params ? `?params=${encodeURIComponent(params)}` : ``
```

Comment:
I don’t think it can? `params` is coming from the app and should always be a valid JS object, while the string to decode is coming from the user (address bar).